virtualmachine/gcp/vm.go

```diff
@@ -56,17 +65,127 @@ type VM struct {
 
 	AccountFile  string
 	account      accountFile
-	SSHCreds     ssh.Credentials // privateKey is required for GCE
+	SSHCreds     ssh.Credentials
 	SSHPublicKey string
+
+	Firewall  string // required when modifying firewall rules
+	Endpoints []Endpoint
+}
+
+// Endpoint represents the protocol and ports configured in a firewall
+type Endpoint struct {
+	// Protocol can be one of the following well known protocol strings
+	// (tcp, udp, icmp, esp, ah, sctp) which are supported by GCP
+	Protocol string
+	Ports    []string
```

Comment:
will it recognise port range given as string?
or for range it has to be sent as list of individual ports only?
e.g.
[port1, port2, port3min-port3max, port4]

Reply:
It will recognize port range.